std/uni/package.d

```diff
@@ -3349,7 +3349,7 @@ struct CowArray(SP=GcPolicy)
         data[$-val.length-1 .. $-1] = val[];
     }
 
-    bool opEquals()(auto const ref CowArray rhs)const
+    bool opEquals()(const auto ref CowArray rhs)const
```

Comment:
Surely `auto ref const` is more appropriate?

Reply:
Why? I feel indifferent about the two.

Reply:
because this is valid syntax: `auto ref const(CowArray) rhs`, and this is not: `const(auto ref CowArray) nhs`.

But in particular, parens like that are _required_ in many cases, for instance:
```
struct S {
  const(T) method();
  const T method(); // not the same; this is a synonym for `T method() const`
}
```
Since using `const(T)` syntax is required for disambiguation in various circumstances, my feeling is that it's much less confusing and certainly more expected to find the `const` immediately before the thing that is const.
If it were up to me, I would even write it in the language spec.

Reply:
It's a conceptual flow thing:
`const auto ref T` goes: type stuff (`const`) -> storage class stuff (`auto ref`) -> type stuff (`T`)
`auto ref const T` goes: storage class stuff (`auto ref`) -> type stuff (`const T`)
It's crude to interleave type stuff and storage class stuff in an arbitrary way.

Reply:
Since it's invalid to put storage class stuff on the end (actual type must be on the end), then I would mandate that storage class stuff should go at the start, not randomly in the middle somewhere.

Reply:
Makes sense, I changed it

Reply:
🤷‍♂️